src/jobs/taskcluster_graph.js

```diff
@@ -92,63 +107,133 @@ export default class TaskclusterGraphJob extends Base {
       host: repositoryUrlParts.host
     };
 
+    let errorGraphUrl =
+      mustache.render(this.config.try.errorTaskUrl, urlVariables);
+
     // Try fetching from .taskgraph.yml, the preferred location, falling back
     // to the project URL The latter is supported only for branches to which
     // .taskcluster.yml (and the task-graph generation support in taskcluster/
     // to which it points) has not yet been merged.
-    let graphUrl, graphText;
     try {
-      graphUrl = projectConfig.tcYamlUrl(this.config.try, urlVariables);
+      let graphUrl = projectConfig.tcYamlUrl(this.config.try, urlVariables);
       console.log(`Fetching '.taskcluster.yml' url ${graphUrl} for '${repo.alias}' push id ${push.id}`);
-      graphText = await fetchGraph(job, graphUrl);
+      let graphText = await fetchGraph(job, graphUrl);
+      templateVariables.source = graphUrl;
+      // Assume .taskcluster.yml has been fetched successfully
+      let queue = new taskcluster.Queue({
+        credentials: this.config.taskcluster.credentials,
+        authorizedScopes: scopes
+      });
+
+      return await this.scheduleTaskGroup(queue,
+                                          repo.alias,
+                                          graphText,
+                                          templateVariables,
+                                          scopes,
+                                          errorGraphUrl);
     } catch (err) {
-      graphUrl = projectConfig.url(this.config.try, repo.alias, urlVariables);
+      let graphUrl = projectConfig.url(this.config.try, repo.alias, urlVariables);
       console.log(`Fetching url ${graphUrl} for '${repo.alias}' push id ${push.id}`);
-      graphText = await fetchGraph(job, graphUrl);
+      let graphText = await fetchGraph(job, graphUrl);
+      templateVariables.source = graphUrl;
+
+      let scheduler = new taskcluster.Scheduler({
+        credentials: this.config.taskcluster.credentials,
+        // Include scopes for creating and extending task graphs.
+        authorizedScopes: scopes.concat(['scheduler:create-task-graph', 'scheduler:extend-task-graph:*'])
+      });
```

Comment:
Do we still need `scheduler`?  It shouldn't be involved in a big-graph-scheduler operation..